android/src/main/java/com/amplitude/android/plugins/AndroidLifecyclePlugin.kt

```diff
@@ -41,18 +41,22 @@ class AndroidLifecyclePlugin : Application.ActivityLifecycleCallbacks, Plugin {
     }
 
     override fun onActivityCreated(activity: Activity, bundle: Bundle?) {
-        if (!hasTrackedApplicationLifecycleEvents.getAndSet(true) && androidConfiguration.defaultTracking.appLifecycles) {
+        @Suppress("DEPRECATION")
+        val trackingAppLifecycle = androidConfiguration.defaultTracking.appLifecycles || androidConfiguration.autocapture.appLifecycles
```

Comment:
Maybe we can have a function to merge the default tracking and autocapture config together, so all the if statements can be shorter.

Reply:
Thanks for the callout! I've tried doing that before, but since it's possible to change defaultTracking options later after the app has initialized Amplitude, merging default tracking and autocapture might discard those changes unintentionally, which would cause a silent bug. For example, if the app does:
```
configs.defaultTracking.sessions = true
```
after initializing Amplitude, there doesn't seem to be a straightforward way to capture this change and update autocapture accordingly.

Reply:
Use one canonical representation of the options internally, probably the auto capture variant. Everything else should bridge to that.

Reply:
124663d0c358e0768c183e18530def4f75efd351 should fix this.

Reply:
> Use one canonical representation of the options internally, probably the auto capture variant. Everything else should bridge to that.

Hi @crleona, I'm leaning more towards a different way of deprecation 
- Keep all old interface (`defaultTracking` and `trackingSessionEvents`) reference unchanged
- Pass `autocapture` value to old interfaces
   - if `autocapture.sessions` is set, set `defaultTracking` and `trackingSessionEvents` accordingly 
   - if `autocapture` is not set, check `defaultTracking` and set `trackingSessionEvents`
It also keeps backward compatible.

Reply:
I implemented a new logic to discard any changes to the `defaultTracking` options and `trackingSessionsEvents` after the configuration is initialized. This is similar to how we're approaching default tracking deprecation in the Swift SDK. Please take a look at b4c09984d035a3b408719a2e06c3d42eb4c2c01a.